### PR TITLE
[Internal] Trigger Assistant by Email

### PR DESCRIPTION
**Pull request by philipperolet**
## Description

Adds ability to trigger conversations with assistant by email using `{assistantName}@run.dust.help`

Ready to ship _internally_ as per discussion in this [thread](https://dust4ai.slack.com/archives/C050A0S2Z7F/p1719322797760439) (admittedly with some delay). 

Goal is to assess whether the feature is useful by recruiting internal users that work with email a lot.
- if useful, this PR _will not_ be ungated as such. Security checks to be added and code to be refactored
- if not useful, this PR is _easy to rollback_ : 2 files separate from the rest of the codebase

The feature is **hard gated** (not feature flagged): only emails sent from `@dust.tt` can trigger assistants.

## Risk
None, gated and changes minor

## Deploy Plan
- deploy `front`
- change sendgrid webhook to correct url
- add EMAIL_SECRET_WEBHOOK to secrets

### CODE REVIEW

File: front/lib/api/assistant/email_trigger.ts [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+    },
+    include: [
+      {
+        model: Conversation,
+      },
+    ],
+    order: [["createdAt", "DESC"]],
+  });
+
+  // TODO: implement good default logic
```
Comment (spolu) [inline front/lib/api/assistant/email_trigger.ts:157]: Can we pin it to our workspace for now?

Comment (spolu) [reply front/lib/api/assistant/email_trigger.ts:157]: That way we can remove the call to ConversationParticipants and make it super simple?

**Review by spolu — COMMENTED** [review]
LGTM but would love to pin it to our workspace for simplicity and safety

File: front/lib/api/assistant/email_trigger.ts (hunk 1 above)
Comment (philipperolet) [reply front/lib/api/assistant/email_trigger.ts:157]: totes :+1:

**Review by spolu — APPROVED** [review]
LGTM

**Pull request merged**